test/integration/scheduler_perf/scheduler_perf_test.go

```diff
@@ -0,0 +1,244 @@
+				}
+
+				op.NodeTemplatePath = &nodeTemplateFile
+			}
+
+			if err := exec.runCreateNodesOp(0 /*opIndex*/, op); err != nil {
+				t.Fatalf("runCreateNodesOp returned error: %v", err)
+			}
+
+			nodes, err := client.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
+			if err != nil {
+				t.Fatalf("listing nodes failed: %v", err)
+			}
+			if got, want := len(nodes.Items), tt.nodeCount; got != want {
+				t.Fatalf("unexpected node count: got %d, want %d", got, want)
+			}
+
+			// Verify node labels if a LabelNodePrepareStrategy was used
+			if tt.nodeStrategy != nil {
+				switch strategy := tt.nodeStrategy.(type) {
+				case *testutils.LabelNodePrepareStrategy:
+					labelValues := make(map[string]bool)
+					for _, v := range strategy.LabelValues {
+						labelValues[v] = false
+					}
+
+					// Check that each node has the expected label with one of the expected values
+					for _, node := range nodes.Items {
+						if labelValue, exists := node.Labels[strategy.LabelKey]; exists {
+							if _, ok := labelValues[labelValue]; ok {
+								labelValues[labelValue] = true
+							} else {
+								t.Errorf("node %s has unexpected label value %s for key %s",
+									node.Name, labelValue, strategy.LabelKey)
+							}
+						} else {
+							t.Errorf("node %s is missing expected label %s", node.Name, strategy.LabelKey)
+						}
+					}
+				case *testutils.UniqueNodeLabelStrategy:
+					// Check that each node has a unique label value for the key
+					labelValues := make(map[string]bool)
+					for _, node := range nodes.Items {
+						if labelValue, exists := node.Labels[strategy.LabelKey]; exists {
+							if labelValues[labelValue] {
+								t.Errorf("node %s has duplicate label value %s for key %s",
+									node.Name, labelValue, strategy.LabelKey)
+							}
+							labelValues[labelValue] = true
+						} else {
+							t.Errorf("node %s is missing expected label %s", node.Name, strategy.LabelKey)
+						}
+					}
+				case *testutils.NodeAllocatableStrategy:
+					// Check that each node has the expected allocatable resources
+					for _, node := range nodes.Items {
+						for resourceName, expectedValue := range strategy.NodeAllocatable {
+							if allocatable, exists := node.Status.Allocatable[resourceName]; !exists {
+								t.Errorf("node %s is missing expected allocatable resource %s",
+									node.Name, resourceName)
+							} else {
+								expectedQuantity := resource.MustParse(expectedValue)
+								if allocatable.Cmp(expectedQuantity) != 0 {
+									t.Errorf("node %s has incorrect allocatable value for %s: got %s, want %s",
+										node.Name, resourceName, allocatable.String(), expectedValue)
+								}
+							}
+						}
+					}
+				default:
+					t.Errorf("No verification implemented for node strategy type: %T", strategy)
+				}
+			}
+		})
+	}
+}
+
+// TestRunCreateNodesOpErrors tests error handling in runCreateNodesOp
+func TestRunCreateNodesOpErrors(t *testing.T) {
```

Comment:
I don't prefer to have a test for error scenario separately. Should be able to include in the above test like adding `expectedErr` or whatever.